File: pi.js

function pi (n){
    sum = 0
    sign = 1
    for (let i = 0; i < n; i++) {
        sum += sign / (2.0*i+1.0)
        sign *= -1;
    }
    return 4.0*sum
}

console.time('test')
console.log(pi(2000000000))
console.timeEnd('test')

File: pi.rs


fn pi(n: i32) -> f64{
    let mut sum :f64 = 0.0;
    let mut sign = 1.0;
    for x in 0..n {
        sum = sum + sign / (2.0*(x as f64)+1.0);
        sign *= -1.0;
    }
    sum * 4.0
}

fn main(){
    let result = pi(2001);
    println!("{}", result);
}
